### ISSUE TITLE
Watch OS Extension: symbol(s) not found for architecture armv7k

### ISSUE THREAD

**Issue by lohithkorp** [Tealium/tealium-ios#68]
Added TealiumWATCHOSExtensionDevicesOnly.framework to my project that has Watch extension and when I try to archive (to submit to the app store), I receive the error **"symbol(s) not found for architecture armv7k"**

I have the latest version downloaded from the github repository.

How do I resolve this?

**Comment by craigrouse** [comment]
@lohithkorp I'm not sure why you're seeing this; the armv7k architecture is targeted in our build process for the DeviceOnly build. I'll get a build running here and see if I can reproduce, but it sounds to me like there may be something wrong with your build settings in the Xcode project. I'll come back and let you know what I find.

**Comment by lohithkorp** [comment]
**sounds to me like there may be something wrong with your build settings in the Xcode project** - could you please suggest me what could have gone wrong, in case if it is not the problem with the build?

**Comment by lohithkorp** [comment]
Any update on this @craigrouse ?

**Comment by lohithkorp** [comment]
We would not be able to proceed for an upcoming release as this is blocking. Could someone look into this issue ASAP please?

**Comment by craigrouse** [comment]
@lohithkorp I am looking into this at the moment. I'll get back to you very shortly.

**Comment by craigrouse** [comment]
@lohithkorp I'm still looking into this. I've been having lots of issues with Xcode; sometimes I'm able to successfully build and archive the app, sometimes not. The correct architecture is definitely present in the DevicesOnly build, however (armv7k), which you can verify with `lipo`.

In any case, the workaround that's working consistently for me is to use the original "TealiumWatchOSExtension.framework" file, and add a script to strip simulator support from the  framework (should only be added to your WatchKit Extension target). Instructions for doing this are here: http://ikennd.ac/blog/2015/02/stripping-unwanted-architectures-from-dynamic-libraries-in-xcode/

Basically, add a new "Run Script" phase, and paste the shell script in there. Rebuild and archive the app, and it should work. If you have trouble doing this, please reply to the support case and we will arrange a call with you.

**Comment by lohithkorp** [comment]
Thanks @craigrouse - I will try this.

**Comment by craigrouse** [comment]
@lohithkorp any luck with this workaround?

**Comment by lohithkorp** [comment]
@craigrouse - I've got very busy with other priorities, but today I could finally test the workaround you provided. It worked like a charm. Thanks for your support. 👍

**Comment by craigrouse** [comment]
Thanks for confirming @lohithkorp. We are working on a fix so that the workaround isn't necessary. I'll let you know when this is available.

**Comment by jonathanswong** [comment]
@lohithkorp We updated our documentation [Tealium for watchOS](https://community.tealiumiq.com/t5/Mobile-Libraries/Tealium-for-watchOS/ta-p/11487) and have a [sample app](https://github.com/Tealium/tealium-ios/tree/master/Samples/WatchOS_Sample_Swift) on Github. Please let us know if this issue is not resolved and we'll reopen it. Thanks.
